pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -50,12 +50,13 @@ const (
 
 // PerconaXtraDBClusterStatus defines the observed state of PerconaXtraDBCluster
 type PerconaXtraDBClusterStatus struct {
-	PXC        AppStatus          `json:"pxc,omitempty"`
-	ProxySQL   AppStatus          `json:"proxysql,omitempty"`
-	Host       string             `json:"host,omitempty"`
-	Messages   []string           `json:"message,omitempty"`
-	Status     AppState           `json:"state,omitempty"`
-	Conditions []ClusterCondition `json:"conditions,omitempty"`
+	PXC               AppStatus          `json:"pxc,omitempty"`
+	ProxySQL          AppStatus          `json:"proxysql,omitempty"`
+	Host              string             `json:"host,omitempty"`
+	Messages          []string           `json:"message,omitempty"`
+	Status            AppState           `json:"state,omitempty"`
+	Conditions        []ClusterCondition `json:"conditions,omitempty"`
+	UpgradeInProgress bool               `json:"upgradeInProgress"`
```

Comment:
Why introduce `UpgradeInProgress`? I believe it's better to extend `AppState` and update `Status`